ambry-router/src/test/java/com.github.ambry.router/NonBlockingRouterTest.java

```diff
@@ -435,30 +444,39 @@ private void testFailureDetectorNotification(OperationHelper opHelper, NetworkCl
     ReplicaId replicaIdToFail =
         indexToFail == -1 ? null : ((RouterRequestInfo) allRequests.get(indexToFail)).getReplicaId();
     for (RequestInfo requestInfo : allRequests) {
+      ResponseInfo responseInfo;
       if (replicaIdToFail != null && replicaIdToFail.equals(((RouterRequestInfo) requestInfo).getReplicaId())) {
-        mockSelectorState.set(MockSelectorState.DisconnectOnSend);
+        responseInfo = new ResponseInfo(requestInfo,
```

Comment:
having the `NetworkClientErrorCode` helps make this consistent - otherwise this is clearly not marking the request as failed.